mesh/mesh.go

```diff
@@ -159,21 +157,10 @@ func NewRecoveredMesh(ctx context.Context, db *DB, atxDb AtxDB, rewardConfig Con
 	if err != nil {
 		logger.With().Panic("failed to load state for layer", msh.LatestLayerInState(), log.Err(err))
 	}
-	// in case we load a state that was not fully played
-	nextLayer := msh.LatestLayerInState().Add(1)
-	lastComplete := msh.trtl.LatestComplete()
-	if nextLayer.Before(msh.trtl.LatestComplete()) {
```

Comment:
say for whatever reason, the rerun on startup yields a different verified layer before the restart. and the verified layer is different from the layer applied layer in state. i think we need to double check that too.

Reply:
that would be a standard revert flow. this can be expected because we may load more data when node is re-started

Reply:
but how about the verified layer is higher than before the restart? say before the restart the state is applied up to N. but after restart, rerun verified up to N+10. how will the state catch up and apply TXs in the extra 10 layers? when the next hare call `HandleIncomingLayer`?

Reply:
for example processed = 20, verified = 10. both processed and verified is loaded from meshdb, so all the data is already there. when we recover start it turns out that we can verify up to 19. so we return old = 10, new = 19 to mesh. and it applies everything in between.

sync will call HandleIncomingLayer from 21 and onward. Hare is more likely from current and onward, and current maybe 21 but unlikely.